src/context/FilterContext/FilterContext.jsx

```diff
@@ -0,0 +1,74 @@
+import { useContext, createContext, useState, useEffect, useReducer } from 'react';
+
+const initialData = {
+	search: '',
+	sortByDate: '',
+	tag: '',
+	priority: '',
+	dataToShow: [],
+	originalData: [],
+};
+
+const FilterContext = createContext();
+
+const FilterProvider = ({ children }) => {
+	const [FilterState, FilterDispatch] = useReducer(filterReducer, initialData);
+	return <FilterContext.Provider value={{ FilterState, FilterDispatch }}>{children}</FilterContext.Provider>;
+};
+
+const useFilter = () => useContext(FilterContext);
+
+export { FilterProvider, useFilter };
+
+const filterReducer = (state, action) => {
+	switch (action.type) {
+		case 'SET_DATA':
+			return {
+				...state,
+				dataToShow: action.payload,
+				originalData: action.payload,
+			};
+		case 'CLEAR_FILTER':
+			return {
+				...state,
+				dataToShow: [...state.originalData],
+				search: state.search,
+				sortByDate: '',
+				tag: '',
+				priority: '',
+			};
+
+		case 'SEARCH':
+			return {
+				...state,
+				search: action.payload,
+			};
+
+		case 'SORT_BY_DATE':
+			if (action.payload === 'OLDEST_TO_NEWEST') {
+				return {
+					...state,
+					sortByDate: 'OLDEST_TO_NEWEST',
+					dataToShow: [...state.dataToShow].sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt)),
```

Comment:
Appreciation: Great that you are spreading the array first as sort is not a pure function.